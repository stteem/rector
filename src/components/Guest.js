import React from 'react';



export default function Guest (props) {

	const [name, setName] = React.useState('');
	const [guest, setGuest] = React.useState('');

	const handleClick = () => {
		setGuest(name);
	}
	// body...
	return(
		<div>
		{ guest !== '' ? <p>My new guest is {guest}</p> : null }
		<input
		  type="text"
          placeholder="Enter guest name"
          onChange={(e) => setName(e.target.value)}
        />
        <button color="primary" onClick={() => handleClick()} >Send Guest</button>
		</div>
	);
}