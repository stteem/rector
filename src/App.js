import React, {Component} from 'react';
import './App.css';
//import Bookings from './components/Bookings';
//import Meals from './components/Meals';
//import Error from './components/Error';
import Guest from './components/Guest';



/*class App extends Component {
    constructor(props){
        super(props);
        this.setGuest = this.setGuest.bind(this);
        this.state ={
            guests: ''

        }
    }
    
    setGuest(obj){
        this.setState({guests: obj});
    }
    

    render() {
        return (
            <div className="container-fluid">
                <center>
                    <h2>Hacker Hostel</h2>
                </center>
                <div className="container">
                    <Bookings setGuest={this.setGuest} />
                    <Error guests={this.state.guests} />
                    <Meals guests={this.state.guests} />
                </div>
            </div>
        );
    }
}

export default App;*/



class App extends Component {
    

  render() {
        return (
            <div className="container-fluid">
                <center>
                    <h2>Guests</h2>
                </center>
                <div className="container">
                    <Guest />
                </div>
            </div>
        );
    }

}

export default App;