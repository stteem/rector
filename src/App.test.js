import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { mount, shallow } from 'enzyme';
import { act, create } from 'react-test-renderer';
import Guest from './components/Guest';

it('renders without crashing', () => {
  shallow(<App />);
});


test('renders Guests', () => {
  const { getByText } = render(<App />);
  const guest = getByText(/Guests/i);
  expect(guest).toBeInTheDocument();
});

it('finds submit button', () => {
  	let wrapper = mount(<App />);
  	expect(wrapper.find('button').text()).toEqual('Send Guest');
});

it('shows input text', () => {
  	let wrapper = mount(<App />);
  	wrapper.find('input').simulate('change', {
		target: {value: 'foo'}
	});
	wrapper.find('button').simulate('click');
	expect(wrapper.containsMatchingElement(<p>My new guest is foo</p>)).toBeTruthy()
});


it('takes snapshot of state changes', () => {

	let component;

	act(() => {
		component = create(<App />);
	});	

	expect(component.toJSON()).toMatchSnapshot();

	//input and clicked
	let instance = component.root;

	let findInput = instance.findByType('input');
	act(() => {
		findInput.props.onChange({ target: { value: 'bar' } });
	});
	
	let findButton = instance.findByType('button');
	act(() => {
		findButton.props.onClick();
	});
	
	expect(component.toJSON()).toMatchSnapshot();
});