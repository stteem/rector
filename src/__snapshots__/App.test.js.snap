// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`takes snapshot of state changes 1`] = `
<div
  className="container-fluid"
>
  <center>
    <h2>
      Guests
    </h2>
  </center>
  <div
    className="container"
  >
    <div>
      <input
        onChange={[Function]}
        placeholder="Enter guest name"
        type="text"
      />
      <button
        color="primary"
        onClick={[Function]}
      >
        Send Guest
      </button>
    </div>
  </div>
</div>
`;

exports[`takes snapshot of state changes 2`] = `
<div
  className="container-fluid"
>
  <center>
    <h2>
      Guests
    </h2>
  </center>
  <div
    className="container"
  >
    <div>
      <p>
        My new guest is 
        bar
      </p>
      <input
        onChange={[Function]}
        placeholder="Enter guest name"
        type="text"
      />
      <button
        color="primary"
        onClick={[Function]}
      >
        Send Guest
      </button>
    </div>
  </div>
</div>
`;
